test/unit/org/apache/cassandra/net/HandshakeTest.java

```diff
@@ -229,7 +239,7 @@ public void testSendIncompatibleOldVersion40() throws InterruptedException
     public void testSendToFuturePost40BelievedToBePre40() throws InterruptedException, ExecutionException
     {
         Result result = handshake(VERSION_30, VERSION_30, current_version, VERSION_30, current_version + 1);
-        Assert.assertEquals(Result.Outcome.SUCCESS, result.outcome);
-        Assert.assertEquals(VERSION_30, result.success().messagingVersion);
+        Assert.assertEquals(Result.Outcome.RETRY, result.outcome);
+        Assert.assertEquals(VERSION_3014, result.retry().withMessagingVersion);
```

Comment:
the handshake will now bump up from requested version 30 to 3014 (so to avoid the columnFilter bug)